tycho-api/src/main/java/org/eclipse/tycho/ReproducibleUtils.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.tycho;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Utility methods for reproducible builds.
+ */
+public class ReproducibleUtils {
+    private ReproducibleUtils() {
+    }
+
+    /**
+     * Writes the property list to the output stream in a reproducible way. The java.util.Properties
+     * class writes the lines in a non-reproducible order, adds a non-reproducible timestamp and
+     * uses platform-dependent new line characters.
+     * 
+     * @param properties
+     *            the properties object to write to the output stream.
+     * @param out
+     *            an output stream.
+     * @param comments
+     *            a description of the property list.
+     * @throws IOException
+     *             if writing the property list to the specified output stream throws an
+     *             IOException.
+     */
+    public static void storeProperties(Properties properties, OutputStream out, String comments) throws IOException {
```

Comment:
1. Do we really use the comments anywhere? If not just drop it.
2. Is there maybe any maven utility or library we can reuse instead of came up with something our self?

Reply:
The comments seem to be used only once, in OsgiSourceMojo. When it creates the pom.properties file, maven-archiver simply drops all the comments (cf. https://github.com/apache/maven-archiver/blob/master/src/main/java/org/apache/maven/shared/archiver/PomPropertiesUtil.java). If it's ok for you I can also drop all the comments, it could greatly simplify the code. I don't know any maven utility to do that, maven-archiver uses a private method for that so I cannot reuse it.

Reply:
I think this comment is not very useful and can be omitted, so I would suggest to go ahead and just drop any comment lines.